### PR TITLE
Pal 324 new infrastructure changes

### PR DESCRIPTION
**Pull request by nw1984**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -85,10 +109,11 @@ podTemplate(containers: [
 
         stage('Maven deploy') {
             dir('Palisade-readers') {
-                container('maven') {
+                container('docker-cmds') {
                     configFileProvider([configFile(fileId: "${env.CONFIG_FILE}", variable: 'MAVEN_SETTINGS')]) {
                         if (("${env.BRANCH_NAME}" == "develop") ||
-                                ("${env.BRANCH_NAME}" == "master")) {
+                                ("${env.BRANCH_NAME}" == "master") ||
+                                ("${env.BRANCH_NAME}" == "PAL-324-new-infrastructure-changes")) {
```
Comment (ac74475) [inline Jenkinsfile:116]: revert this change so there isn't a mention of the feature branch name

File: Jenkinsfile [hunk 2]
```diff
@@ -13,10 +13,35 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
```
Comment (ac74475) [inline Jenkinsfile:15]:
```suggestion
 */
 
//node-affinity
//nodes 1..3 are reserved for Jenkins slave pods.
//node 0 is used for the Jenkins master
```

File: Jenkinsfile [hunk 3]
```diff
@@ -13,10 +13,35 @@
...
+          matchExpressions:
+          - key: palisade-node-name
+            operator: In
+            values: 
+            - node1
+            - node2
+            - node3
+  containers:
+  - name: docker-cmds
+    image: 779921734503.dkr.ecr.eu-west-1.amazonaws.com/jnlp-did:INFRA
```
Comment (ac74475) [inline Jenkinsfile:34]: Does this need to specify the ECR? If so can we parameterise this?

Comment (nw1984) [reply Jenkinsfile:37]: Since we are using a real Kubernetes installation rather than microk8s, it is necessary to specify the full tag for the ECR.
I have tried parameterising the variable using ${}, ${env.}, $(), $(env.) - none of which result in the substitution working at all.
The intention, is to incorporate the pod template definition into the Jenkins template longer term so that the Jenkinsfiles are sparse, so the visibility of the ECR registry should not remain an issue.
I can spend more time looking at this but it is not as obvious as it would first appear.
